src/workbench/connect-dialogs/TransferHexDialog.tsx

```diff
@@ -0,0 +1,145 @@
+ *
+ * SPDX-License-Identifier: MIT
+ */
+import { Button } from "@chakra-ui/button";
+import {
+  Flex,
+  HStack,
+  Icon,
+  Image,
+  Link,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+import { FormattedMessage } from "react-intl";
+import { GenericDialog } from "../../common/GenericDialog";
+import transferHexWin from "./transfer-hex-win.gif";
+import transferHexMac from "./transfer-hex-mac.gif";
+import { ReactNode } from "react";
+import { RiExternalLinkLine } from "react-icons/ri";
+import { useDeployment } from "../../deployment";
+
+export const enum TransferHexChoice {
+  CancelDontShowAgain,
+  Cancel,
+}
+
+interface TransferHexDialogProps {
+  callback: (value: TransferHexChoice) => void;
+  dialogNormallyHidden: boolean;
+}
+
+export const TransferHexDialog = ({
+  callback,
+  dialogNormallyHidden,
+}: TransferHexDialogProps) => {
+  return (
+    <GenericDialog
+      onClose={() => callback(TransferHexChoice.Cancel)}
+      body={<TransferHexDialogBody />}
+      footer={
+        <TransferHexDialogFooter
+          dialogNormallyHidden={dialogNormallyHidden}
+          onCancel={() => callback(TransferHexChoice.Cancel)}
+          onCancelDontShowAgain={() =>
+            callback(TransferHexChoice.CancelDontShowAgain)
+          }
+        />
+      }
+      size="3xl"
+    />
+  );
+};
+
+const TransferHexDialogBody = () => {
+  const isMac = /Mac/.test(navigator.platform);
+  const { guideLink } = useDeployment();
+  return (
+    <VStack
+      width="auto"
+      ml="auto"
+      mr="auto"
+      p={5}
+      pb={0}
+      spacing={5}
+      alignItems="flex-start"
+    >
+      <VStack alignItems="flex-start">
+        <Text as="h2" fontSize="xl" fontWeight="semibold">
+          <FormattedMessage id="transfer-hex-title" />
+        </Text>
+        <Text>
+          <FormattedMessage
+            id="transfer-hex-message"
+            values={{
+              strong: (chunks: ReactNode) => (
+                <Text as="span" fontWeight="semibold">
+                  {chunks}
+                </Text>
+              ),
+            }}
+          />
+        </Text>
+        <Text>
+          <FormattedMessage
+            id="transfer-hex-support-message"
+            values={{
+              link: (chunks: ReactNode) => (
+                <Link
+                  color="brand.500"
+                  target="_blank"
+                  rel="noreferrer"
+                  href={guideLink}
+                >
+                  {chunks}{" "}
+                  <Icon as={RiExternalLinkLine} verticalAlign="middle" />
+                </Link>
+              ),
+            }}
+          />
+        </Text>
+      </VStack>
+      <Flex justifyContent="center" width="100%">
+        {isMac ? (
+          <Image width="100%" height="100%" src={transferHexMac} alt="" />
+        ) : (
+          <Image width="100%" height="100%" src={transferHexWin} alt="" />
+        )}
```

Comment:
Might be nicer to determine the src with a branch and only have one Image tag?

I think no alt is OK given that the sentence above is explaining the drag.

Reply:
Yes, that's sensible. This was leftover from when width and height values were different for each. [`3f05893`](https://github.com/microbit-foundation/python-editor-next/pull/791/commits/3f05893beb09ad020a91f341d50800055cdbc313).